File: LAB3/src/shoeTypes/HikingBoots.java
package shoeTypes;

import enums.Brand;
import enums.Size;
import enums.Type;
import shoe.Shoe;

public class HikingBoots extends Shoe {
    private String soleMaterial;

    public HikingBoots(String name, Type type, Brand brand, Size size, String color, double price, String material, String soleMaterial) {
        super(name, type, brand, size, color, price, material);
        this.soleMaterial = soleMaterial;
    }

    public String getSoleMaterial() {
        return soleMaterial;
    }

    public void setSoleMaterial(String soleMaterial) {
        this.soleMaterial = soleMaterial;
    }
}

File: lab11/shoes/HikingBoots.py
from shoes.Shoes import *
from enums.Type import Type


class HikingBoots(Shoes):
    type = Type.SPORT

    def __init__(self, name, brand, size, price, colour):
        self.name = name
        self.brand = brand
        self.size = size
        self.price = price
        self.colour = colour

    def __str__(self):
        return "Name: " + self.name + " Brand " + str(self.brand.name) + " Size " + str(self.size) + " Colour: " + str(
            self.colour) + " Type: " + str(self.type.name) + " Price " + str(self.price)
